modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/type/Type.java

```diff
@@ -39,22 +42,82 @@
   boolean isAssignableFrom(Class<?> clazz);
 
   /**
-   * Checks the assignability of the current type to the given class
+   * Checks the assignability of the current type from the given {@link Type}
+   *
+   * @param type The type to check
+   * @return a boolean indicating whether the type is assignable or not to the given {@link Type}
+   * @since 4.1
+   */
+  boolean isAssignableFrom(Type type);
+
+  /**
+   * Checks the assignability of the current type to the given {@link Class}
    *
    * @param clazz The class to check
    * @return a boolean indicating whether the type is assignable or not to the given class
    * @since 4.1
    */
   boolean isAssignableTo(Class<?> clazz);
 
+  /**
+   * Checks the assignability of the current type to the given {@link Type}
+   *
+   * @param type The type to check
+   * @return a boolean indicating whether the type is assignable or not to the given {@link Type}
+   * @since 4.1
+   */
+  boolean isAssignableTo(Type type);
+
+  /**
+   * Checks equality of the current type to the given {@link Type}
+   *
+   * @param type The type to check equality
+   * @return a boolean indicating whether the type is the same type or not
+   * @since 4.1
+   */
+  boolean isSameType(Type type);
+
+  /**
+   * Checks equality of the current type to the given {@link Class}
+   *
+   * @param clazz The type to check equality
+   * @return a boolean indicating whether the type is the same type or not
+   * @since 4.1
+   */
+  boolean isSameType(Class<?> clazz);
+
+  /**
+   * @return a boolean indicating if the current type is instantiable or not.
+   * @since 4.1
+   */
+  boolean isInstantiable();
+
   /**
    * @return The generics for the current type.
    * @since 4.1
    */
   List<TypeGeneric> getGenerics();
 
-  //TODO: Remove once the type loader exists MULE-14040
-  java.lang.reflect.Type getReflectType();
+  /**
+   * @param interfaceClass The {@link Class} with generics
+   * @return The list of generics types from the given interface class.
+   * @since 4.1
+   */
+  List<Type> getInterfaceGenerics(Class interfaceClass);
```

Comment:
why interface and not just "getGenerics"? some reason this won't work con `class`?

Reply:
I'm just reusing and exposing a method that already existed in `IntrospectionUtils`